gremlin-dotnet/src/Gremlin.Net/Structure/IO/GraphSON/TimeSpanConverter.cs

```diff
@@ -0,0 +1,45 @@
+﻿#region License
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Xml;
+using Newtonsoft.Json.Linq;
+
+namespace Gremlin.Net.Structure.IO.GraphSON
+{
+    internal class TimeSpanConverter : IGraphSONSerializer, IGraphSONDeserializer
+    {
+        public Dictionary<string, dynamic> Dictify(dynamic objectData, GraphSONWriter writer)
+        {
+            TimeSpan value = objectData;
+            return GraphSONUtil.ToTypedValue("Duration", XmlConvert.ToString(value), "gx");
```

Comment:
`XmlConvert.ToString()` uses ISO-8601 representation (ie `PT8H6M12.345S`)?

I see no info on the MSDN: https://msdn.microsoft.com/en-us/library/z1955e3x

We are using `PTnHnMnS` as defined by java Duration: https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html#toString--

Reply:
Seems to be implicit by how XML treats durations: https://www.w3.org/TR/xmlschema-2/#duration. The output of XmlConvert.ToString() follows the PTnHnMnS format.

Reply:
Oh, nice TIL!